src/compiler/transformers/classFields.ts

```diff
@@ -632,7 +723,21 @@ namespace ts {
             // From ES6 specification:
             //      HasLexicalDeclaration (N) : Determines if the argument identifier has a binding in this environment record that was created using
             //                                  a lexical declaration such as a LexicalDeclaration or a ClassDeclaration.
-            const staticProperties = getProperties(node, /*requireInitializer*/ true, /*isStatic*/ true);
+            let staticProperties = getProperties(node, /*requireInitializer*/ true, /*isStatic*/ true);
+
+            if (languageVersion === ScriptTarget.ESNext) {
+                // We can't initialize static private properties after the class as the are not in the lexical scope
+                // so we rely on standard init even under useDefineForClassFields:false
+                staticProperties = filter(staticProperties, p => !isPrivateIdentifier(p.name));
+            }
+            else {
+                // We must initialize static private properties even if the don't have a value as we initialize them with a wrapper object which mist always exist.

```

Comment:
I find this logic a bit counter-intuitive. It looks like initialized and unitialized private static fields should be transformed the same way but because public fields are transformed differently we need do this dance. I left a question below about the wrapper object. If it's not needed, it would simplify some stuff here. If needed, I wonder if we couldn't transform private section in a more uniform way.

Reply:
I think the wrapper object is necessary. Unless we inline the assignment everywhere..
```ts
_C_x = __classStaticPrivateFieldSet(C, C, 10);
```